system-apps/desktop-documentation-viewer/webClient/src/App.tsx

```diff
@@ -0,0 +1,168 @@
+      destination: ZoweZLUX.uriBroker.pluginRESTUri(this.props.resources.pluginDefinition.getBasePlugin(), 'plugindetail', "")
+    };
+  }
+
+  updateOrInitState(stateUpdate: any): void {
+    if (!this.state) {
+      this.state = Object.assign(this.getDefaultState(), stateUpdate);
+    }
+    else {
+      this.setState(stateUpdate);
+    }
+  }
+
+  handleLaunchOrMessageObject(data: any) {
+    switch (data.type) {
+    case 'setAppRequest':
+      let actionType = data.actionType;
+      let msg:string;
+      if (actionType == 'Launch' || actionType == 'Message') {
+        let mode = data.targetMode;
+        if (mode == 'PluginCreate' || mode == 'PluginFindAnyOrCreate') {
+          this.updateOrInitState({actionType: actionType,
+                                  appTarget: mode,
+                                  appId: data.targetAppId,
+                                  parameters: data.requestText});
+        } else {
+          msg = `Invalid target mode given (${mode})`;
+          this.log.warn(msg);
+          this.updateOrInitState({status: msg});
+        }
+      } else {
+        msg = `Invalid action type given (${actionType})`;
+        this.log.warn(msg);
+        this.updateOrInitState({status: msg});
+      }
+      break;
+    default:
+      this.log.warn(`Unknown command (${data.type}) given in launch metadata.`);
+    }
+  }
+
+  zluxOnMessage(eventContext: any): Promise<any> {
+    return new Promise((resolve,reject)=> {
+      if (eventContext != null && eventContext.data != null && eventContext.data.type != null) {
+        resolve(this.handleLaunchOrMessageObject(eventContext.data));
+      } else {
+        let msg = 'Event context missing or malformed';
+        this.log.warn('onMessage '+msg);
+        return reject(msg);
+      }
+    });
+  }
+  
+  provideZLUXDispatcherCallbacks(): ZLUX.ApplicationCallbacks {
+    return {
+      onMessage: (eventContext: any): Promise<any> => {
+        return this.zluxOnMessage(eventContext);
+      }      
+    }
+  }  
+
+  getPluginsFromServer() {
+    let allPlugins = ZoweZLUX.pluginManager.loadPlugins("application", false);
+    this.updateOrInitState({allPlugins: allPlugins.__zone_symbol__value})
+  }
+
+  getPluginDocsInfoFromServer(identifier: string) {
+    let url = ZoweZLUX.uriBroker.pluginRESTUri(this.props.resources.pluginDefinition.getBasePlugin(), 'plugindetail', 'docs/'+`${identifier}`);
+    fetch(url, {
+      method: 'GET'
+    })
+      .then(res => {
+        this.log.info(`Res=${res}`);
+        res.json()
+          .then((responseJson) => {
+            if (responseJson) {
+              this.setState({
+                pluginIdentifier: identifier,
+                docsList: responseJson.doc
+              });
+            }
+          })
+      });
+  }
+
+  getDoc(identifier: string, path: string) {
+    let url = ZoweZLUX.uriBroker.pluginRESTUri(this.props.resources.pluginDefinition.getBasePlugin(), 'plugindetail', `${identifier}`);
```

Comment:
Looks like youre posting an identifier and location to this URL, so method name is misleading

Reply:
Changed it to `postDocToServer`.